src/implementations/nlp/IncrementalVocabulary.ts

```diff
@@ -1,12 +1,25 @@
-import { IIncrementalVocabulary } from "interfaces";
-import { Word, WordRoot, VocabularySize, WordIndex } from "objects";
-import { WordToIndex } from "objects/business/nlp/WordToIndex";
+import { Word, WordRoot, WordIndex, DuplicateWordError } from "crunchDB/objects";
+import { IIncrementalVocabulary } from "crunchDB/interfaces"; 
 import { Vocabulary } from "./Vocabulary";
+import {  okAsync, errAsync,  ResultAsync } from 'neverthrow'; 
 
 export class IncrementalVocabulary extends Vocabulary implements IIncrementalVocabulary {
-    public addWord(word: Word | WordRoot): void {
-        this.wordToIndex[word] = WordIndex(this.indexToWord.length);
-        this.indexToWord.push(word);
+
+  /*
+  * @param words: an array of words/ word roots
+  * @description: adds words/ word roots to the vocabulary
+  * @throws Err if the word/ word root is already present in the vocabulary
+  * */
+  public addWord(word: Word | WordRoot): ResultAsync<void, DuplicateWordError> {
+    if (this.wordToIndex.has(word)) { // duplicate check 
+      return errAsync( new DuplicateWordError("Word already exists in the vocabulary")); 
+    }
+    else{
+      this.wordToIndex.set(word, WordIndex(this.wordToIndex.size));
+      this.indexToWord.push(word); 
+      return okAsync(void 0);
```

Comment:
You can `return okAsync(undefined)` to meet a "void" return type, that's what we do everywhere else.